packages/components/src/components/carousel/carousel.test.stories.ts

```diff
@@ -0,0 +1,277 @@
+    (story: () => typeof html) => html`
+      <style>
+        sd-carousel {
+          padding: 0.5px;
+        }
+      </style>
+      ${story()}
+    `
+  ]
+};
+
+/**
+ * This shows sd-carousel in its default state.
+ */
+
+export const Default = {
+  name: 'Default',
+  render: (args: any) => {
+    return generateTemplate({
+      args
+    });
+  }
+};
+
+/**
+ * Use the `variant` attribute to select the pagination format.
+ */
+
+export const Variant = {
+  name: 'Variant',
+  parameters: { controls: { exclude: 'variant' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'variant', values: ['dot', 'number'] }
+      },
+      args
+    });
+  }
+};
+
+/**
+ * Use the `inverted` attribute to make a carousel with inverted colors.
+ */
+
+export const Inverted = {
+  name: 'Inverted',
+  parameters: { controls: { exclude: ['inverted', 'variant'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: { type: 'attribute', name: 'variant' },
+        y: { type: 'attribute', name: 'inverted' }
+      },
+      args,
+      constants: [{ type: 'attribute', name: 'autoplay', value: true }],
+      options: {
+        templateBackgrounds: { alternate: 'y', colors: ['rgb(var(--sd-color-primary, 0 53 142))', 'white'] }
+      }
+    });
+  }
+};
+
+/**
+ * Use the `loop` attribute to enable/disable the looping of your slides.
+ */
+
+export const Loop = {
+  name: 'Loop',
+  parameters: { controls: { exclude: 'loop' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'loop' }
+      },
+      args
+    });
+  }
+};
+
+/**
+ * Use the `autoplay` attribute to toggle autoplay.
+ */
+
+export const Autoplay = {
+  name: 'Autoplay',
+  parameters: { controls: { exclude: 'autoplay' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'autoplay' }
+      },
+      constants: [
+        { type: 'attribute', name: 'variant', value: 'dot' },
+        { type: 'attribute', name: 'loop', value: 'true' }
+      ],
+      args
+    });
+  }
+};
+
+/**
+ * Use the `slides-per-page` attribute to set the number of slides that would be shown at a given time.
+ */
+
+export const SlidesPerPage = {
+  name: 'Slides Per Page',
+  parameters: { controls: { exclude: 'slides-per-page' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'slides-per-page', values: [1, 2, 3] }
+      },
+      constants: [{ type: 'attribute', name: 'variant', value: 'dot' }],
+      args
+    });
+  }
+};
+
+/**
+ * Use `slides-per-move` to set how many slides the carousel advances when scrolling. This is useful when specifying a `slides-per-page` greater than one. By setting `slides-per-move` to the same value as `slides-per-page`, the carousel will advance by one page at a time.<br>
```

Comment:
Use the `slides-per-move` attribute  to configure the number of slides the carousel scrolls through at a time.

Reply:
Updated